### PR TITLE
chore: [M3-7803] - Clean up `Chip` component

### PR DESCRIPTION
**Pull request by bnussman-akamai**
## Description 📝

After talking about Chips in our cafe meeting, I looked at the Chip component and I couldn't un-see some of this code 😩😤

This PR simplifies our override of MUI's `<Chip />` and removes some _styled components_ that went way too far

## Preview 📷

> [!note]
> The goal of this PR is to clean up Chip related code, but **not change to how it looks** 👁️  

| Before | After |
|--------|-------|
|   Screenshot 2024-02-22 at 4 06 20 PM     |  Screenshot 2024-02-22 at 4 06 48 PM     |
|     Screenshot 2024-02-22 at 4 07 56 PM   |     Screenshot 2024-02-22 at 4 08 19 PM  |
| Screenshot 2024-02-22 at 4 09 56 PM | Screenshot 2024-02-22 at 4 09 33 PM  |

## How to test 🧪

- Check out this PR
- Check that the Chip works as expected in Storybook
- Some examples of Chips you could check are:
  - Kubernetes HA chip (visible on Kubernetes landing and details page) 💻 
  - Database HA chip (visible on the Database landing page) 💾 
  - Verify the Linode Status on a Linode's details page looks and works as expected 🟢 (This use to be a chip, now it is just a status icon and text)
  - Check the VPC subnet IPs chip
- Check any other instance of `Chip` you can think of in Cloud Manager

## As an Author I have considered 🤔

- [ ] 👀 Doing a self review
- [ ] ❔ Our [contribution guidelines](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md)
- [ ] 🤏 Splitting feature into small PRs
- [x] ➕ Adding a changeset
- [ ] 🧪 Providing/Improving test coverage
- [ ] 🔐 Removing all sensitive information from the code and PR description
- [ ] 🚩 Using a feature flag to protect the release
- [ ] 👣 Providing comprehensive reproduction steps
- [ ] 📑 Providing or updating our documentation
- [ ] 🕛 Scheduling a pair reviewing session
- [ ] 📱 Providing mobile support
- [ ] ♿  Providing accessibility support

### CODE REVIEW

**Review by jaalah-akamai — APPROVED** [review]
Nice cleanup 🧹

File: packages/manager/src/features/Kubernetes/ClusterList/KubernetesClusterRow.tsx [hunk 1]
```diff
@@ -107,32 +107,31 @@ export const KubernetesClusterRow = (props: Props) => {
...
+            {hasUpgrade && (
               <Chip
                 clickable
-                inTable
                 label="UPGRADE"
                 onClick={openUpgradeDialog}
                 size="small"
               />
-            ) : null}
+            )}
```
Comment (dwiley-akamai) [inline packages/manager/src/features/Kubernetes/ClusterList/KubernetesClusterRow.tsx:134]: The "Upgrade" chip was adversely impacted by some of the styling changes:

| Prod  | This branch |
| ------------- | ------------- |
| Screenshot 2024-02-23 at 11 42 11 AM | Screenshot 2024-02-23 at 11 42 07 AM |

File: packages/manager/src/features/Linodes/LinodeEntityDetailHeader.tsx [hunk 2]
```diff
@@ -146,26 +149,27 @@ export const LinodeEntityDetailHeader = (
...
+            {formattedStatus}
+          </Typography>
+        </Stack>
         {isRebootNeeded && (
           <TooltipIcon
             status="help"
             sxTooltipIcon={{ padding: 0 }}
             text={VPC_REBOOT_MESSAGE}
           />
         )}
```
Comment (dwiley-akamai) [inline packages/manager/src/features/Linodes/LinodeEntityDetailHeader.tsx:171]: Similar to the K8S "Upgrade" chip situation:

| Prod | This branch |
| ------------- | ------------- |
| Screenshot 2024-02-23 at 11 45 59 AM | Screenshot 2024-02-23 at 11 45 56 AM |

**Review by dwiley-akamai — CHANGES_REQUESTED** [review]
The cleanup looks good mostly but I noticed a couple of instances where chips were adversely impacted

File: packages/manager/src/features/Kubernetes/ClusterList/KubernetesClusterRow.tsx (hunk 1 above)
Comment (bnussman-akamai) [reply packages/manager/src/features/Kubernetes/ClusterList/KubernetesClusterRow.tsx]: Fixed in 63bb02e35a

 Thank you for catching this!

| Before Fix  | Fixed |
|--------|-------|
|   Screenshot 2024-02-23 at 12 12 27 PM     |  Screenshot 2024-02-23 at 12 12 20 PM     |

File: packages/manager/src/features/Linodes/LinodeEntityDetailHeader.tsx (hunk 2 above)
Comment (bnussman-akamai) [reply packages/manager/src/features/Linodes/LinodeEntityDetailHeader.tsx]: Fixed in 63bb02e

**Review by dwiley-akamai — APPROVED** [review]

**Pull request merged**